Workbooks/Azure Active Directory/EntitlementManagementAppRoleAssignments/EntitlementManagementAppRoleAssignments.workbook

```diff
@@ -0,0 +1,105 @@
+{
+  "version": "Notebook/1.0",
+  "items": [
+    {
+      "type": 1,
+      "content": {
+        "json": "## Application role assignment activity"
+      },
+      "name": "text - 2"
+    },
+    {
+      "type": 9,
+      "content": {
+        "version": "KqlParameterItem/1.0",
+        "parameters": [
+          {
+            "id": "a5151809-64d6-45ef-8881-499597a4727c",
+            "version": "KqlParameterItem/1.0",
+            "name": "Since",
+            "label": "Activity since last",
+            "type": 2,
+            "description": "Include audit events since a particular date",
+            "isRequired": true,
+            "value": "30d",
+            "typeSettings": {
+              "additionalResourceOptions": []
+            },
+            "jsonData": " [ {\"value\":\"1d\",\"label\":\"24 hours\"}, {\"value\":\"7d\",\"label\":\"7 days\"}, {\"value\":\"30d\",\"label\":\"30 days\"}, {\"value\":\"90d\",\"label\":\"90 days\"}, { \"value\":\"3653d\", \"label\":\"all\" }]\r\n",
```

Comment:
why is this using a custom thing for time range and doing it at all as text instead of using literal time range parameters?

Reply:
there are many reasons to use a time range parameter, some for simplicity, it can simplify the query, as yo udon't need to do ` TimeGenerated > ago({Since:value})`, you can do literally  `TimeGenerated {Since}` if you want the content in the query and have support for common time ranges and custom time ranges like "tuesday 5:35 pm to wednesday 9:55am".  if you select the time range parameter in the step's time range setting, you don't even need to reference the time range in the query text itself, the time range will be passed as a standard logs query api parameter.

using a time range parameter also means if a user were to pin a query using this time range to an azure dashboard, the query will use the dashboard's time range and will auto refresh on the dashboard.

Reply:
updated to new time range parameter type